src/python/nxt_python_wsgi.c

```diff
@@ -692,10 +719,16 @@ static int
 nxt_python_add_sptr(nxt_python_ctx_t *pctx, PyObject *name,
     nxt_unit_sptr_t *sptr, uint32_t size)
 {
-    char      *src;
-    PyObject  *value;
+    return nxt_python_add_char(pctx, name, nxt_unit_sptr_get(sptr), size);
+}
+
 
-    src = nxt_unit_sptr_get(sptr);
+static int
+nxt_python_add_char(nxt_python_ctx_t *pctx, PyObject *name,
+    char *src, uint32_t size)
+{
+    int       res;
+    PyObject  *value;
 
     value = PyString_FromStringAndSize(src, size);
     if (nxt_slow_path(value == NULL)) {
```

Comment:
This seems like an unrelated refactoring.  Please separate it into a commit that goes before the main change, to be able to better discern the real changes.